pygama/dsp/_processors/template.py

```diff
@@ -0,0 +1,77 @@
+# 1) Import python modules
+#
+import numpy as np
+from numba import guvectorize
+
+# 2) Provide instructions to numba
+# 
+# Documentation about numba guvectorize decorator:
+# https://numba.pydata.org/numba-doc/latest/user/vectorize.html#the-guvectorize-decorator
+#
+# Notes:
+# - use `nopython=True` and `cache=True`         (@ian, is this correct?)
+# - use 64-bit variables, e.g. int64 and float64 (@all, this is up for discussion)
+# - use [:] for al output parameters             (@ian, is this correct?)
+#
+@guvectorize("void(float64[:], float64, int64, float64[:], int64[:])",
+             "(n),(),()->()", nopython=True, cache=True)
+
+# 3) Define the processor interface
+#
+# Notes: (@all, this is up for discussion)
+# - add the `_in`/`_out` suffix to the name of the input/output variables
+# - use w_ for waveforms, t_ for indexes, a_ for amplitudes 
+# - use underscore casing for the name of the processor and variables, e.g.
+#   a_trap_energy_in or t_trigger_in
+#
+def the_processor_template(w_in, t_in, a_in, w_out, t_out):
+
+    # 4) Document the algorithm
+    #
+    """
+    Add here a complete description of what the processor does, including the
+    meaning of input and output variables
+    """
+
+    # 5) Initialize output parameters (@iann, will this be done automatically by the processing chain?)
```

Comment:
The problem with trying to initialize the output parameters is that a lot of unexpected behavior will still overwrite these values. For example, if you feed fixed_time_pickoff an index that is out of range, but not by enough to cause a segfault, it will overwrite it with a nonsense value. I think it is a better idea to explicitly handle the cases where we want to use a NaN value within the processor

Reply:
Alright, if everybody is fine with that I will implement initialize the outputs at the beginning of each processor